### PR TITLE
Define the agent fingerprint as the SPKI.

### PR DESCRIPTION
**Pull request by mfoltzgoogle**
Addresses Issue #282: Certificates should have a maximum lifetime, and SPAKE2 identities should be SPKI not cert fingerprint

This defines the agent fingerprint in terms of the SPKI (e.g., the public key) of the agent certificate.  The agent fingerprint is used to identify agents and is input to SPAKE2.

This allows agents to use shorter-lived certificates and renew their certificates without having to re-authenticate to other agents.

This PR does not specify certificate lifetimes or renewal policies.  It also does not handle the issuing of completely new certificates while maintaining previous trust relationships.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/openscreenprotocol/pull/301.html" title="Last updated on Jan 22, 2024, 8:26 PM UTC (596e9f3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/openscreenprotocol/301/15ed7a5...596e9f3.html" title="Last updated on Jan 22, 2024, 8:26 PM UTC (596e9f3)">Diff</a>

### CODE REVIEW

**Comment by mfoltzgoogle** [comment]
This should address the issues raised in #282, while overall certificate management needs more discussion.

**Comment by mfoltzgoogle** [comment]
@baylesj to take a look.

File: index.bs [hunk 1]
```diff
@@ -322,6 +318,19 @@ it should change the DNS-SD service name to a new value, indicating a new
 mechanism for metadata discovery.
 
 
+Computing the Agent Fingerprint {#computing-agent-fingerprint}
+-------------------------------
+
+The <dfn>agent fingerprint</dfn> of an agent is computed by following these
+steps:
+
+1. Compute the [[RFC7469#section-2.4|SKPI Fingerprint]] of the [=agent certificate=]
```
Comment (baylesj) [inline index.bs:327]: Super nit: In some other areas of index.bs an HTML list is used for describing processes. Should we standardize on one or the other?

File: index.bs [hunk 2]
```diff
@@ -322,6 +318,19 @@ it should change the DNS-SD service name to a new value, indicating a new
 mechanism for metadata discovery.
 
 
+Computing the Agent Fingerprint {#computing-agent-fingerprint}
+-------------------------------
```
Comment (baylesj) [inline index.bs:322]: Standardize on section breaks?
```
=======================================================
```
or 
```
-------------------------------
```

File: index.bs [hunk 3]
```diff
@@ -290,12 +290,8 @@ Advertising agents must include DNS TXT records with the following
...
 
 : fp
-:: The <dfn>agent fingerprint</dfn> of the advertising agent, computed over the
-    [=agent certificate=]. The format of the fingerprint is defined by [[!RFC5122]],
-    excluding the "fingerprint:" prefix and including the hash function, space,
-    and hex-encoded fingerprint.  The fingerprint value also functions as an ID
-    for the agent. All agents must support the following hash functions: [=sha-256=],
-    [=sha-512=].  Agents must not support the following hash functions: [=md2=], [=md5=].
+:: The [=agent fingerprint=] of the advertising agent. The steps to compute
+    the agent fingerprint are defined below.
```
Comment (baylesj) [inline index.bs:294]: Since it's not defined immediately below, consider adding a section reference.

File: index.bs [hunk 4]
```diff
@@ -322,6 +318,19 @@ it should change the DNS-SD service name to a new value, indicating a new
...
 
 
+Computing the Agent Fingerprint {#computing-agent-fingerprint}
+-------------------------------
+
+The <dfn>agent fingerprint</dfn> of an agent is computed by following these
+steps:
+
+1. Compute the [[RFC7469#section-2.4|SKPI Fingerprint]] of the [=agent certificate=]
+    according to [[!RFC7469]] using [[RFC6234|SHA-256]] as the hash algorithm.
```
Comment (baylesj) [inline index.bs:328]: Is this "The Pin Directive" described in that spec?

https://datatracker.ietf.org/doc/html/rfc7469#section-2.1.1

**Review by baylesj — APPROVED** [review]
LGTM % feedback.

File: index.bs (hunk 1 above)
Comment (mfoltzgoogle) [reply index.bs:329]: Markdown will convert this into an HTML `<ol>` for us, unless it's nested inside a manually-added HTML element.

File: index.bs (hunk 2 above)
Comment (mfoltzgoogle) [reply index.bs:324]: These for different heading levels (see https://commonmark.org/help/).  If you prefer the # based syntax I guess we could change.

File: index.bs (hunk 3 above)
Comment (mfoltzgoogle) [reply index.bs:292]: The `[=agent fingerprint=]` syntax will link directly to the definition, do you think something else is needed?

File: index.bs (hunk 4 above)
Comment (mfoltzgoogle) [reply index.bs:330]: Not exactly.  This is just referring to the SPKI which the pin directive consumes.  The pin directives are for use in HTTP server headers to pin certificates to hosts.

**Comment by mfoltzgoogle** [comment]
@backkem, this PR uses the SPKI as defined in RFC 7469, but thank you for the earlier PR :-)

**Comment by mfoltzgoogle** [comment]
@baylesj, PTAL

**Comment by mfoltzgoogle** [comment]
Went and merged this as I believe the last comments are addressed.  Please let me know if there are additional comments @baylesj

**Pull request merged**